src/netlink/nl_bond.cc

```diff
@@ -297,4 +319,42 @@ int nl_bond::update_lag_member(rtnl_link *old_slave, rtnl_link *new_slave) {
   return 0;
 }
 
+int nl_bond::add_l3_address(rtnl_link *link) {
+  int rv = 0;
+#ifdef HAVE_RTNL_LINK_BOND_GET_MODE
+  assert(link);
+
+  std::deque<rtnl_addr *> addresses;
+  nl->get_l3_addrs(link, &addresses);
+
+  for (auto i : addresses) {
+    rv = nl->add_l3_addr(i);
+    if (rv < 0)
+      LOG(ERROR) << __FUNCTION__ << ":failed to add l3 address " << i << " to "
+                 << OBJ_CAST(link);
+  }
+
+#endif
+  return rv;
+}
+
+int nl_bond::remove_l3_address(rtnl_link *link) {
+  int rv = 0;
+#ifdef HAVE_RTNL_LINK_BOND_GET_MODE
+  assert(link);
+
+  std::deque<rtnl_addr *> addresses;
+  nl->get_l3_addrs(link, &addresses);
+
+  for (auto i : addresses) {
+    rv = nl->del_l3_addr(i);
+    if (rv < 0)
+      LOG(ERROR) << __FUNCTION__ << ":failed to add l3 address to "
```

Comment:
```suggestion
      LOG(ERROR) << __FUNCTION__ << ":failed to remove l3 address from "
```

And maybe print the address?